### PR TITLE
grid layout

### PR DESCRIPTION
**Pull request by apryka**
changes related to react-grid-layout implementation

### CODE REVIEW

File: test/components/ColorPicker.test.js [hunk 1]
```diff
@@ -47,15 +47,15 @@ describe('<ColorPicker />', () => {
...
-    it('should call onChange once value is changed', () => {
-      wrapper.find('input').simulate('change', { target: { value: changedColor }});
-      expect(onChange).toHaveBeenCalled();
-    });
-
-    it('should change input value after change', () => {
-      wrapper.find('input').simulate('change', { target: { value: changedColor }});
-      expect(wrapper.find('input').prop('value')).toEqual(changedColor.toUpperCase());
-    });
+    // it('should call onChange once value is changed', () => {
```
Comment (maciejrybaniec) [inline test/components/ColorPicker.test.js:50]: Are you planing to restore that tests in the future ?

File: lib/components/FormInput.js [hunk 2]
```diff
@@ -9,25 +9,29 @@ const FormInput = ({
...
-  value,
+  value = '',
   error,
   children,
   label,
   ...props
 }) => {
-  const [inputValue, setInputValue] = useState('');
+  let timer = null;
+  const [inputValue, setInputValue] = useState(value);
```
Comment (maciejrybaniec) [inline lib/components/FormInput.js:18]: Why you need to store value as a state in that component ? Especially when it comes from properties - there is two source of truth here.

Can't we remove local state and base only on `value` from props ?

File: lib/components/ColorPicker.js [hunk 3]
```diff
@@ -41,12 +42,15 @@ const ColorPicker = props => {
...
   };
   const handleChange = color => {
-    if (!color.hex) {
-      setColor({ hex: initialColor });
-      return;
-    }
-    setColor(color);
-    onChange(color, title);
+    clearTimeout(timer);
+    timer = setTimeout(() => {
```
Comment (maciejrybaniec) [inline lib/components/ColorPicker.js]: Few questions to this code.
1. Why we need setTimeout and pushing `setColor` to end of event loop ?
2. Is `500` some kind of magic number here :) ?

**Review by maciejrybaniec — CHANGES_REQUESTED** [review]

**Comment by maciejrybaniec** [comment]
Please - squash all commits.

File: test/components/ColorPicker.test.js (hunk 1 above)
Comment (apryka) [reply test/components/ColorPicker.test.js:50]: all tests will be re-written once final shape of theme builder is known. I'll create separate issue for tests

File: lib/components/ColorPicker.js (hunk 3 above)
Comment (apryka) [reply lib/components/ColorPicker.js]: debounce removed, it's not necessary here

File: lib/components/FormInput.js (hunk 2 above)
Comment (apryka) [reply lib/components/FormInput.js:18]: performance issue in dashboard builder. If we rely on value from props we need to wait till all changes are applied to the dashboard and then update the input. 

I map value from prop to state on component mount event and treat the state as a single source of truth.

File: test/components/ColorPicker.test.js (hunk 1 above)
Comment (maciejrybaniec) [reply test/components/ColorPicker.test.js:50]: 👍

**Review by maciejrybaniec — APPROVED** [review]

**Pull request merged**